example/main.cpp

```diff
@@ -147,49 +47,122 @@ void algo(std::shared_ptr<ioh::problem::bbob::bbob_base> f, std::shared_ptr<ioh:
 		l->do_log(f->loggerCOCOInfo());
 		best_y = std::max(y, best_y);
 	}
-	
 }
 
 
-class Base
-{
-public:
-	virtual void x() { print("Base"); }
-};
-
+// ioh::suite::bbob bbob;
+// std::shared_ptr<ioh::problem::bbob::bbob_base> problem;
+// while ((problem = bbob.get_next_problem()) != nullptr)
+// 	std::cout << problem->get_problem_name() << std::endl;
+//
+//
+// std::cout << "*****************" << std::endl;
+// ioh::suite::bbob bbob({1}, {1, 2}, {5, 6});
+// std::shared_ptr<ioh::problem::bbob::bbob_base> problem;
+// while ((problem = bbob.get_next_problem()) != nullptr)
+// 	std::cout << problem->get_problem_name() << std::endl <<
+// 		problem->get_instance_id() << std::endl <<
+// 		problem->get_number_of_objectives() << std::endl;
+//
+//
+// std::vector<int> p{1};
+// std::vector<int> i{2};
+// std::vector<int> d{4};
+//
+// register_in_factory<BaseOneArg, C, int> c("C");
+// register_in_factory<BaseOneArg, D, int> d("D");
+// std::cout << factory<BaseOneArg, int>::get().create("D", 1)->get_name() << std::endl;
 
-class Derived: public Base
+int main()
 {
-public:
-	void x() override { print("Derived"); }
-};
 
+	using namespace ioh::problem;
+	using namespace ioh::logger;
 
+	size_t sample_size = 100;
 
-class DerivedDerived : public Derived
-{
-public:
-	void x() override { print("DerivedDerived"); }
-};
+	std::vector<int> pbs = { 1,2 };
+	std::vector<int> ins = { 1,2 };
+	std::vector<int> dims = { 2,10 };
+	
+	
+	ioh::suite::bbob bench(pbs, ins, dims);
+	size_t ecdf_width = 20;
+	using Logger = ecdf<bbob::bbob_base>;
+	
+	range_log<double> error(0, 6e7, ecdf_width);
+	range_log<size_t> evals(0, sample_size, ecdf_width);
+	Logger logger(error, evals);
+	
+	logger.activate_logger();
+	logger.track_suite(bench);
+	
+	
+	size_t seed = 5;
+	std::mt19937 gen(seed);
+	std::uniform_real_distribution<> dis(-5, 5);
```

Comment:
Dear Johann,

Thank you so much having taken the time to review this. I totally understand that is to much to look at all the commits directly, and I agree we should take a second look at anything I might have missed during this refractoring. 

W.r.t to the random module, I feel this still requires a major overhaul. In this PR, it tried to keep as many code as-is, in the future we definetly will look into this. The same holds for the experimenter constructor taking a concrete csv logger implementation, I agree this should be changed, but is beyond the scope of this PR.

I will also change the nameing scheme of the constants in the defines file. I agree, we might also would like to rename the file itself.  The hint conserning the use of macros in the random module is also very apreciated. 

Thanks again,
Jacob

Reply:
I understand the difficulty of scoping a refactoring, that's why I put most comments in a separated issue, sorry if I mixed-up things a bit. Anyway, the cumbersome part being done, it will be easier to propose more atomic PRs in the future.